src/Microsoft.AspNetCore.Razor.Language/Legacy/ImplicitExpressionEditHandler.cs

```diff
@@ -105,57 +105,57 @@ protected override PartialParseResult CanAcceptChange(Span target, TextChange no
             }
 
             // Accepts cases when insertions are made at the end of a span or '.' is inserted within a span.
-            if (IsAcceptableInsertion(target, normalizedChange))
+            if (IsAcceptableInsertion(target, change))
             {
                 // Handle the insertion
-                return HandleInsertion(target, lastChar.Value, normalizedChange);
+                return HandleInsertion(target, lastChar.Value, change);
             }
 
-            if (IsAcceptableDeletion(target, normalizedChange))
+            if (IsAcceptableDeletion(target, change))
             {
-                return HandleDeletion(target, lastChar.Value, normalizedChange);
+                return HandleDeletion(target, lastChar.Value, change);
             }
 
             return PartialParseResult.Rejected;
         }
 
         // A dotless commit is the process of inserting a '.' with an intellisense selection.
-        private static bool IsDotlessCommitInsertion(Span target, TextChange change)
+        private static bool IsDotlessCommitInsertion(Span target, SourceChange change)
         {
             return IsNewDotlessCommitInsertion(target, change) || IsSecondaryDotlessCommitInsertion(target, change);
         }
 
         // Completing 'DateTime' in intellisense with a '.' could result in: '@DateT' -> '@DateT.' -> '@DateTime.' which is accepted.
-        private static bool IsNewDotlessCommitInsertion(Span target, TextChange change)
+        private static bool IsNewDotlessCommitInsertion(Span target, SourceChange change)
         {
             return !IsAtEndOfSpan(target, change) &&
-                   change.NewPosition > 0 &&
```

Comment:
@NTaylorMullen - what's the **meaning** of `NewPosition`?? I just can't get my head around why this code was correct.